src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/PerformanceAnalyzerApp.java

```diff
@@ -107,21 +107,26 @@ public static void main(String[] args) {
         (MetricsConfiguration.CONFIG_MAP.get(StatsCollector.class).samplingInterval) / 2,
         TimeUnit.MILLISECONDS);
     PluginSettings settings = PluginSettings.instance();
-    StatsCollector.STATS_TYPE = "agent-stats-metadata";
-    METRIC_COLLECTOR_EXECUTOR.addScheduledMetricCollector(StatsCollector.instance());
-    StatsCollector.instance().addDefaultExceptionCode(StatExceptionCode.READER_RESTART_PROCESSING);
-    METRIC_COLLECTOR_EXECUTOR.setEnabled(true);
-    METRIC_COLLECTOR_EXECUTOR.start();
-
-    final GRPCConnectionManager connectionManager = new GRPCConnectionManager(
-        settings.getHttpsEnabled());
-    final ClientServers clientServers = createClientServers(connectionManager, appContext);
-    startErrorHandlingThread(THREAD_PROVIDER, exceptionQueue);
-
-    startReaderThread(appContext, THREAD_PROVIDER);
-    startGrpcServerThread(clientServers.getNetServer(), THREAD_PROVIDER);
-    startWebServerThread(clientServers.getHttpServer(), THREAD_PROVIDER);
-    startRcaTopLevelThread(clientServers, connectionManager, appContext, THREAD_PROVIDER);
+    if (ConfigStatus.INSTANCE.haveValidConfig()) {
+      StatsCollector.STATS_TYPE = "agent-stats-metadata";
+      METRIC_COLLECTOR_EXECUTOR.addScheduledMetricCollector(StatsCollector.instance());
+      StatsCollector.instance()
+          .addDefaultExceptionCode(StatExceptionCode.READER_RESTART_PROCESSING);
+      METRIC_COLLECTOR_EXECUTOR.setEnabled(true);
+      METRIC_COLLECTOR_EXECUTOR.start();
+
+      final GRPCConnectionManager connectionManager =
+          new GRPCConnectionManager(settings.getHttpsEnabled());
+      final ClientServers clientServers = createClientServers(connectionManager, appContext);
+      startErrorHandlingThread(THREAD_PROVIDER, exceptionQueue);
+
+      startReaderThread(appContext, THREAD_PROVIDER);
+      startGrpcServerThread(clientServers.getNetServer(), THREAD_PROVIDER);
+      startWebServerThread(clientServers.getHttpServer(), THREAD_PROVIDER);
+      startRcaTopLevelThread(clientServers, connectionManager, appContext, THREAD_PROVIDER);
+    } else {
+      LOG.error("Performance analyzer app stopped due to invalid config status.");
```

Comment:
Does the process also exit with error at this point or is it just stalled? We should exit with error.